map/booking_filter_processor.cpp

```diff
@@ -0,0 +1,76 @@
+#include "map/booking_filter_processor.hpp"
+#include "map/booking_availability_filter.hpp"
+
+#include "search/result.hpp"
+
+namespace booking
+{
+namespace filter
+{
+FilterProcessor::FilterProcessor(Index const & index, booking::Api const & api)
+  : m_index(index)
+  , m_api(api)
+{
+  m_filters.emplace(Type::Deals, std::make_unique<AvailabilityFilter>(*this));
+  m_filters.emplace(Type::Availability, std::make_unique<AvailabilityFilter>(*this));
+}
+
+void FilterProcessor::ApplyFilters(search::Results const & results,
```

Comment:
1. Can we have a test for this method?
2. I have a feeling that keeping the results in a variable and running the filters in the normal order (as in, with a loop instead of chained callbacks) will be more readable.

Reply:
Я планировал написать тесты когда добавлю хотя бы еще один фильтр - горячие предложения (в следующем пулл реквесте).
Вызывать просто в цикле не получается, потому что ApplyFilter содержит асинхронный вызов по сети. Можно было бы конечно засинхронизировать, но тогда возникла бы еще одна проблема - файловый поток мог бы подвисать на одну большую и тяжелую задачу (особенно при плохом интернете).